openshift/console
# OCPBUGS-17341: OCP console mandate secret for repository creation

**Fixes**: 
https://issues.redhat.com/browse/OCPBUGS-17341

**Analysis / Root cause**: 
The UX of the Form raised confusion among users over making the Secret field mandatory. It was reported that when using the CLI, the secret was not required.

**Solution Description**: 
Due to the addition of a new feature that allows the automatic attachment of webhooks to the Git Repositories, the Secret field is now mandatory. Have added a helpText explaining the need in the form.
Also removed the action of opening the Advanced Settings Expandable menu manually, as the hidden fields are pretty much mandatory. From now on, the Hidden fields will be automatically shown once the user has provided a valid Git URL.

**Screenshots / Gifs for design review**: 

https://github.com/openshift/console/assets/47265560/11d7c7cf-927f-4249-9b6d-0a11c5c10494

**Unit test coverage report**: 
Not changed.

**Test setup:**
1. Install Pipelines operator
2. Open the Pipeline Repositories Form and create a repository with Git URL: https://github.com/Lucifergene/oc-pipe
3. Shall see the Hidden fields automatically pop up and a help text popup beside the Secret Label.



**Browser conformance**:
